pkg/prow/prow.go

```diff
@@ -628,11 +629,8 @@ func (o *Options) GetPostSubmitJob(org, repo, branch string) (config.Postsubmit,
 
 	key := fmt.Sprintf("%s/%s", org, repo)
 	for _, p := range prowConfig.Postsubmits[key] {
-
-		for _, a := range p.Branches {
-			if a == branch {
-				return p, nil
-			}
+		if p.Brancher.ShouldRun(branch) {
```

Comment:
So this actually panics atm:

```
--- FAIL: TestGetPostSubmitJob (0.01s)
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x52293f]
goroutine 65 [running]:
testing.tRunner.func1(0xc0004db600)
	/usr/local/go/src/testing/testing.go:830 +0x392
panic(0x1890040, 0x2d00ef0)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
regexp.(*Regexp).doExecute(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1af91ea, 0x6, 0x0, 0x0, ...)
	/usr/local/go/src/regexp/exec.go:527 +0x6f
regexp.(*Regexp).doMatch(...)
	/usr/local/go/src/regexp/exec.go:514
regexp.(*Regexp).MatchString(...)
	/usr/local/go/src/regexp/regexp.go:468
k8s.io/test-infra/prow/config.Brancher.ShouldRun(0x0, 0x0, 0x0, 0xc0006e0400, 0x1, 0x4, 0x0, 0x0, 0x1af91ea, 0x6, ...)
	/go/pkg/mod/k8s.io/test-infra@v0.0.0-20190131093439-a22cef183a8f/prow/config/jobs.go:222 +0x146
github.com/jenkins-x/**/pkg/prow.(*Options).GetPostSubmitJob(0xc00055bd08, 0x1af71d8, 0x4, 0x1af7108, 0x4, 0x1af91ea, 0x6, 0x0, 0x0, 0x0, ...)
	/workspace/source/pkg/prow/prow.go:632 +0x400
github.com/jenkins-x/**/pkg/prow_test.TestGetPostSubmitJob(0xc0004db600)
	/workspace/source/pkg/prow/prow_test.go:491 +0x42a
testing.tRunner(0xc0004db600, 0x1ba46f0)
	/usr/local/go/src/testing/testing.go:865 +0xc0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:916 +0x35a
```

I think the problem is in this code of `Brancher`:
```
	if len(br.Branches) == 0 || br.re.MatchString(branch) {
		return true
	}
```
Looks like `br.re` is nil.

Reply:
yes, I forgot to call the func that compiles the regexes.